chatmeleon-server/prisma/schema.prisma

```diff
@@ -56,12 +56,17 @@ model Account {
 
 // Define the Conversation model
 model Conversation {
-    id            String   @id @default(auto()) @map("_id") @db.ObjectId
-    createdAt     DateTime @default(now())
+    id        String   @id @default(auto()) @map("_id") @db.ObjectId
+    createdAt DateTime @default(now())
+    name      String?
+
     lastMessageAt DateTime @default(now())
-    name          String?
-    isGroup       Boolean?
+    lastMessage   String?
+
+    isOriginal  Boolean?
+    groupAvatar String?
 
+    nicknames   Json?
```

Comment:
From my understanding, `{ "a": 36, "b": ["dheil", 47, null], "c": null }` is also a valid JSON. What I mean is, it would be great if theres a way to define an object type that maps a String to a String instead of a json that can be anything. Of course, I don’t know if there’s such feature in prisma so it’s just a suggestion to look into. Hope that’s clear enough

Reply:
https://www.prisma.io/docs/orm/prisma-schema/data-model/models#defining-composite-types

There seems to be this thing called composite type. Why don’t we just define `type UserNickname {userId String; nickname String}` and the field in conversation being `nicknames UserNickname[]`?

That would be a more definite type and nothing can go wrong. Possibly

Reply:
And somehow define the userId in the nickname list to be unique as well…

Reply:
That makes sense, Thank you i will look into it